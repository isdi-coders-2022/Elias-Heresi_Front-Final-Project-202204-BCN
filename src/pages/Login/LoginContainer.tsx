import styled from "styled-components";

export const LoginContainer = styled.div`
  height: 100vh;
  background-color: rgb(111, 219, 135);
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  h1 {
    padding: 15px 0;
  }
`;
