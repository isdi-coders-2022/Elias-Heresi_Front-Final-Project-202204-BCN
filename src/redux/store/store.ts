import { configureStore } from "@reduxjs/toolkit";
import userReducer from "../features/userSlice";
import uiReducer from "../features/uiSlice";
import diaryReducer from "../features/diarySlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
    diary: diaryReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
