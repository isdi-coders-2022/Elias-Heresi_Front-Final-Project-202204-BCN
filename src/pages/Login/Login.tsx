import { useSelector } from "react-redux";
import LoginForm from "../../components/LoginForm/LoginForm";
import { RootState } from "../../redux/store/store";
import Loading from "../../components/Loading/Loading";
import { Ui } from "../../redux/interfaces/UiInterface";
import { useEffect } from "react";
import { UserState } from "../../redux/interfaces/UserInterface";
import { useNavigate } from "react-router-dom";
import { RegisterContainer } from "../Register/RegisterContainer";
import { ToastContainer } from "react-toastify";

const Login = (): JSX.Element => {
  const { loading }: Ui = useSelector((state: RootState) => state.ui);
  const { logged }: UserState = useSelector((state: RootState) => state.user);
  const navigate = useNavigate();

  useEffect(() => {
    if (logged) {
      navigate("/home");
    }
  }, [navigate, logged]);

  return (
    <>
      {loading && <Loading />}
      <ToastContainer />
      <RegisterContainer>
        <img
          src="https://images.pexels.com/photos/708392/pexels-photo-708392.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
          alt="Placeholder"
          className="form__image"
        />
        <div className="login__form">
          <h1>Bonanza</h1>
          <h2>Login to your account</h2>
          <LoginForm />
        </div>
      </RegisterContainer>
    </>
  );
};

export default Login;
