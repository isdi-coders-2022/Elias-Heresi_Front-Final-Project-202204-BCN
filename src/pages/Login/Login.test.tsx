import { Provider } from "react-redux";
import store from "../../redux/store/store";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import Login from "./Login";

const mockUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUseNavigate,
}));

let mockLogged: boolean = false;

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: () => ({ logged: mockLogged }),
}));

describe("Given the Login page component", () => {
  describe("When it's invoked and the user isn't logged in", () => {
    test("Then it should render an image and 2 headings", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Login />
          </Provider>
        </BrowserRouter>
      );

      const searchedHeaders = screen.getAllByRole("heading");
      const searchedImages = screen.getAllByRole("img");

      const expectedHeaders = 2;
      const expectedImages = 1;

      expect(searchedHeaders).toHaveLength(expectedHeaders);
      expect(searchedImages).toHaveLength(expectedImages);
    });
  });
  describe("If the user is logged in", () => {
    test("Then the user will be redirected to another page", () => {
      mockLogged = true;

      render(
        <BrowserRouter>
          <Provider store={store}>
            <Login />
          </Provider>
        </BrowserRouter>
      );

      expect(mockUseNavigate).toHaveBeenCalled();
    });
  });
});
