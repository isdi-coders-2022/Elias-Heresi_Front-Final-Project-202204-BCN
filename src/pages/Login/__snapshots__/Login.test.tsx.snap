// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Login page component When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-gsnTZi ftoCxN"
>
  <img
    alt="Placeholder"
    className="login__image"
    src="https://csbs.utah.edu/_resources/images/Happy%20People.jpg"
  />
  <div
    className="login__form"
  >
    <h1>
      Bonanza
    </h1>
    <h2>
      Login to your account
    </h2>
    <div
      className="sc-bczRLJ brPyMg"
    >
      <form
        autoComplete="off"
        className=""
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Enter your username"
            type="text"
            value=""
          />
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
            value=""
          />
        </div>
        <section
          className="container text-center"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            type="submit"
          >
            Submit
          </button>
          <p>
            Not a member yet?
          </p>
          <a
            className="btn btn-secondary"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </section>
      </form>
    </div>
  </div>
</div>
`;
